src/tests/ftest/aggregation/aggregation_basic.py

```diff
@@ -0,0 +1,126 @@
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+
+import time
+from ior_test_base import IorTestBase
+
+class DaosAggregationBasic(IorTestBase):
+    # pylint: disable=too-many-ancestors
+    """Test class Description:
+
+       Run IOR with same file option to verify the
+       aggregation happens in the background. Check for the
+       space usage before and after the aggregation to verify
+       the overwritten space is reclaimed.
+
+    :avocado: recursive
+    """
+
+    def test_basic_aggregation(self):
+        """Jira ID: DAOS-3451
+
+        Test Description:
+            Purpose of this test is to run ior using DFS api to write to
+            the DAOS and rerun it again with the same file option and
+            enable the aggregation and verify the aggregation reclaims
+            the overwritten space.
+
+        Use case:
+            Run ior with read, write, CheckWrite, CheckRead
+            Run ior with read, write, CheckWrite, CheckRead with same file
+            Enable aggregation and wait for 90 seconds to let the files
+            old enough to be marked for aggregation.
+            Wait for aggregation to complete.
+            Check the utilized capacity of the container/pool. It should be
+            roughly the same as after the initial run since aggregation has
+            reclaimed the overwritten capacity.
+
+        :avocado: tags=all,pr,hw,large,aggregate,daosio,aggregatebasic
+        """
+
+        # Create pool and container
+        self.update_ior_cmd_with_pool()
+
+        # Since the transfer size is 1M, the objects will be inserted
+        # directly into NVMe and hence storage_index = 1
+        storage_index = 1
+
+        pool_info = self.pool.pool.pool_query()
+        initial_free_space =\
+            pool_info.pi_space.ps_space.s_free[storage_index]
+
+        # Disable the aggregation
+        self.pool.set_property("reclaim", "disabled")
+
+        # Run ior with -k option to retain the file and not delete it
+        self.run_ior_with_pool()
+        pool_info = self.pool.pool.pool_query()
+        free_space_after_first_ior =\
+            pool_info.pi_space.ps_space.s_free[storage_index]
+
+        space_used_by_ior = initial_free_space - free_space_after_first_ior
+
+        self.log.info("Space used by first ior = {}".format(
```

Comment:
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments